File: KTP/contests/codeforces/solo_01_02_2023/A/tsk.cpp
#include <iostream>

using namespace std;
#define ull unsigned long long

int main(){
    int n; cin >> n;
    char s; 
    int num;

    ull sum = 0;
    int cnt = 0;

    for(int i = 0; i < n; i++){
        cin >> s >> num;
        if(s == '+'){
            sum += num; cnt ++;
            cout.precision(15);
            if(sum == 0) cout << 0 << '\n';
            else cout << (double)sum / cnt << '\n';

        }
        else{
            sum -= num; cnt--;
            cout.precision(15);
            if(cnt == 0){
                cout << 0 << '\n';
            }
            else{
                cout << (double)sum / cnt << '\n';
            }
        }
    }
    return 0;
}

File: KTP/contests/codeforces/solo_01_02_2023/A/tsk.py
t = int(input())
sum = 0
cnt = 0
for _ in range(t):
    s, num = input().split()
    num = int(num)
    if(s == '+'):
        sum += num
        cnt += 1
        print(sum / cnt)
    else:
        sum -= num
        cnt -= 1
        if(cnt == 0):
            print(0)
        else:
            print(sum / cnt)
